pkg/client/helper.go

```diff
@@ -349,9 +356,22 @@ func TransportFor(config *Config) (http.RoundTripper, error) {
 	var transport http.RoundTripper
 	if config.Transport != nil {
 		transport = config.Transport
-	} else {
-		if tlsConfig != nil {
-			transport = &http.Transport{
+	} else if tlsConfig != nil {
```

Comment:
Extract this to a function-- will make it more obvious that the cache is of *unwrapped* transports. Also will constrain the scope of the lock more appropriately.